DataConnectors/Agari/AzureFunctionAgari/run.ps1

```diff
@@ -83,6 +91,37 @@ Function Post-LogAnalyticsData($customerId, $sharedKey, $body, $logType)
     return $response.StatusCode
 }
 
+# Timer Function to stamp the last successful log into the environment variable
+Function SetLastLogTime ($subid,$resGrp,$appName,$nextStartDate,$apdLogSuccess,$aprLogSuccess,$bpLogSuccess)
+{
+    #Get the subscription
+    Select-AzSubscription -Subscriptionid $subid
+    #Getting WebApp info
+    $webApp = Get-AzWebApp -ResourceGroupName $resGrp -Name $appName
+
+    # Read App Settings and load into Hash Table
+    $appSettingList = $webApp.SiteConfig.AppSettings
+    $newAppSettingList = @{}
+    ForEach ($kvp in $appSettingList) {
+        $newAppSettingList[$kvp.Name] = $kvp.Value
+    }
+
+    # Add in new value for last log time NOTE: have to mangle the date with a "t" prefix due to issue Set-AzWebApp #8277
+    if ($apdLogSuccess){
+        $newAppSettingList['APDlastLogTime'] = "t$nextStartDate"
+    }
+    if ($aprLogSuccess){
+        $newAppSettingList['APRlastLogTime'] = "t$nextStartDate"
+    }
+    if ($bpLogSuccess){
+        $newAppSettingList['BPlastLogTime'] = "t$nextStartDate"
+    }
+    # Write to App Settings if there are changes
+    if(($apdLogSuccess) -or ($aprLogSuccess) -or ($bpLogSuccess)){
+        Set-AzWebApp -ResourceGroupName $resGrp -Name $appName -AppSettings $newAppSettingList
+    }
```

Comment:
If we get a 4xx error, the invoke-restapi won't create any psobjects so the foreach loop won't trigger thus the logsuccess won't flip to true. The next run will fetch from whatever time is in the environment variable, so it won't have any gaps in the logs. if it does run, we want to populate the env variable with the next start date. I have this running in sentinel now, with our agari data, and it's been running successfully for the last week.  We need to get this to MS so that we can get the private preview where we can watch / assess with 2 customers.  In the future we should consider putting a "api last accessed field in the APIs so we don't have to rely on the app logging fails / success data requests.